src/data/data_fetcher.py

```diff
@@ -273,3 +276,30 @@ def get_transactions_on_page(self):
         self.last_contribution_receipt_date = self.info["pagination"]["last_indexes"][
             "last_contribution_receipt_date"
         ]
+
+    def _build_df(self):
+        self.df = pd.DataFrame(
+            self.complete_list,
+            columns=[
+                "contributor_occupation",
+                "contributor_employer",
+                "contributor_city",
+                "contributor_state",
+                "contributor_zip",
+                "party",
+            ],
+        )
+        self.df.fillna(value="", inplace=True)
+        return self.df
+
+    def _save_df_data(self):
+        did_write = False
+        pattern = f"*_{self.recipient_committee_type}_in_{self.two_year_transaction_period}.csv"
+        files = os.listdir("data/")
+        for name in files:
+            if fnmatch.fnmatch(name, pattern):
+                os.remove("data/" + name)
+                self.df.to_csv(f'data/{self.pages_pulled}_of_{self.total_pages}_for_{self.recipient_committee_type}_in_{self.two_year_transaction_period}.csv')
+                did_write = True
+        if did_write != True:
+            self.df.to_csv(f'data/{self.pages_pulled}_of_{self.total_pages}_for_{self.recipient_committee_type}_in_{self.two_year_transaction_period}.csv') 
```

Comment:
Do we need `did_write`? Can't we just `df.to_csv()` at the end and call it a day?

Writing in the loop doesn't make sense to me-- if we had `7/8/9/10_of_X_for_H_in_2020.csv`, we'd save 4 copies of the same file before the function finished executing.

Reply:
`pattern = f"*_{self.recipient_committee_type}_in_{self.two_year_transaction_period}.csv"` 
My idea of this is that you will never have more than one file with the same `pattern`, so you couldn't have 7/8/9/10 because when you go to save your next file it will delete the old one. 


And yeah I guess did_write is useless, my brain thinks that it should write two files but they are named the same thing so it doesn't work how I was thinking. 🤔 🧐